src/main/java/us/kbase/mobu/ModuleBuilder.java

```diff
@@ -27,669 +27,704 @@
-		jc.addCommand(VERSION_COMMAND, versionArgs);
-
-		// add the 'install' command
-		InstallCommandArgs installArgs = new InstallCommandArgs();
-		jc.addCommand(INSTALL_COMMAND, installArgs);
-
-		// add the 'run' command
-		RunCommandArgs runArgs = new RunCommandArgs();
-		jc.addCommand(RUN_COMMAND, runArgs);
-
-		// parse the arguments and gracefully catch any errors
-		try {
-			jc.parse(args);
-		} catch (RuntimeException e) {
-			showError("Command Line Argument Error", e.getMessage());
-			System.exit(1);
-		}
+	public static final String DEFAULT_METHOD_STORE_URL =
+			"https://appdev.kbase.us/services/narrative_method_store/rpc";
+
+	public static final String VERSION = "0.1.0";
+	
+	// keep a single source of truth for the version info
+	private static final String VERSION_INFO =
+			"KBase SDK+ version " + VERSION + " (commit " + GitCommit.COMMIT + ")";
+
+	public static void main(final String[] args) throws Exception {
+		final int exitCode = new CommandLine(new ModuleBuilder()).execute(args);
+		System.exit(exitCode);
+	}
 
-		// if the help flag is set, then show some help and exit
-		if(gArgs.help) {
-			showBriefHelp(jc, System.out);
-			return;
-		}
+	@Override
+	public void run() {
+		// fallback when no subcommand is used
+		CommandLine.usage(this, System.out);
+	}
 
-		// no command entered, just print brief info and exit
-		if(jc.getParsedCommand()==null) {
-			showBriefHelp(jc, System.out);
-			return;
-		}
+	public static class Verbose {
 
-		// if we get here, we have a valid command, so process it and do stuff
-		int returnCode = 0;
-		if(jc.getParsedCommand().equals(HELP_COMMAND)) {
-			showCommandUsage(jc,help,System.out);
+		@Option(
+				names = {"-v", "--verbose"},
+				description = "Show verbose output including stack traces when appropriate.",
+				defaultValue = "false"
+		)
+		boolean verbose;
+	}
+	
+	public static class ValidationArgs extends Verbose {
+		
+		// TODO UX this doesn't actually make much sense. I tried to summarize but the code
+		//         is kind of bonkers, just read the ModuleValidator constructor.
+		//         It should probably be removed.
+		//         Changing validate to take only one module made it somewhat less crazy
+		@Option(
+				paramLabel = "<method_store_url>",
+				names = {"-m", "--method_store"},
+				description = """
+						The URL of the KBase Narrative Method Store to use when validating method \
+						specifications.\
+						""",
+				defaultValue = ModuleBuilder.DEFAULT_METHOD_STORE_URL,
+				showDefaultValue = CommandLine.Help.Visibility.ALWAYS
+		)
+		String methodStoreUrl;
+
+		// TODO UX seems like this could be removed, not sure it's still relevant.
+		//         just looks for a specific value in the spec.json file. Check w/ Bill
+		@Option(
+				names = {"-a", "--allow_sync_method"},
+				description = "Allow synchronous methods (advanced).",
+				defaultValue = "false"
+		)
+		boolean allowSyncMethods;
```

Comment:
Synchronous methods? I'm not even sure what that means here. What does it look for in spec.json? Should probably remove it.